packages/permissionless/accounts/kernel/signerToEcdsaKernelSmartAccount.ts

```diff
@@ -210,6 +358,7 @@ export type SignerToEcdsaKernelSmartAccountParameters<
     address?: Address
     index?: bigint
     factoryAddress?: Address
+    metaFactoryAddress?: Address
     accountLogicAddress?: Address
     ecdsaValidatorAddress?: Address
```

Comment:
should we have kernel version as a param?

Reply:
The implementation is tied to v2.2 and v3.0 strictly mostly.  Kept v2.2 implementation since users might be using it already as it was already present in the SDK. Although there's two new 2.x versions `2.3` and `2.4` after that which doesn't have major changes just some feature support. And since `Kernel v3` supports all the previous version feature anyway just did Kernel v3 PR and jump over those two "unsupported" versions.

This is also the reason  didn't include `kernelVersion` param. But let me think about it.

Reply:
> Also if I am not wrong users could have just changes the factory address and used the v2.3 or v2.4 of kernel because they way user op was constructed has not changed right?

Yes sure with our SDKs that's true. With permissionless not likely.

So these were the changes between Kernel version updates:

v2.1 => v2.2, we removed delegate call support from execute function, and native batching function was added
v2.2 => v2.3, we added executeDelegatecall for explicit delegatecall function, added eip1271 sig wrapper (similar to `wrapMessageHash` in Kernel v3)
v2.3 => v2.4, we had minor changes on eip1271 function to allow modular permission

So to support `v2.3` and `v2.4` we have to add the relevant `wrapMessageHash` util. It's bit different than `Kernel v3`. And as `delegateCall` is not supported anyway it's irrelevant.

Reply:
@plusminushalf  I think if all good. Let's merge this now. And if needed can add support for `v2.3` and `v2.4` later in separate PR.